src/parsing.jl

```diff
@@ -0,0 +1,260 @@
+
+function Base.start(pqv::PQValue)
+    sv = string_view(pqv)
+    return (sv, start(sv))
+end
+
+function Base.next(pqv::PQValue, state)
+    sv, sv_state = state
+    c, new_sv_state = next(sv, sv_state)
+    return c, (sv, new_sv_state)
+end
+
+function Base.done(pqv::PQValue, state)
+    sv, sv_state = state
+    return done(sv, sv_state)
+end
+
+## integers
+_DEFAULT_TYPE_MAP[:int2] = Int16
+_DEFAULT_TYPE_MAP[:int4] = Int32
+_DEFAULT_TYPE_MAP[:int8] = Int64
+
+## floating point
+_DEFAULT_TYPE_MAP[:float4] = Float32
+_DEFAULT_TYPE_MAP[:float8] = Float64
+
+## oid
+_DEFAULT_TYPE_MAP[:oid] = Oid
+
+## numeric
+_DEFAULT_TYPE_MAP[:numeric] = Decimal
+
+# no default for monetary; needs lconv and lc_monetary from result/connection
+
+## character
+# bpchar is char(n)
+Base.parse(::Type{String}, pqv::PQValue{PQ_SYSTEM_TYPES[:bpchar]}) = rstrip(pqv, ' ')
+# char is "char"
+_DEFAULT_TYPE_MAP[:char] = PQChar
+Base.parse(::Type{PQChar}, pqv::PQValue{PQ_SYSTEM_TYPES[:char]}) = PQChar(first(pqv))
+# varchar, text, and name are all String
+
+## binary data
+
+_DEFAULT_TYPE_MAP[:bytea] = Vector{UInt8}
+function Base.parse(::Type{Vector{UInt8}}, pqv::PQValue{PQ_SYSTEM_TYPES[:bytea]})
+    byte_length = Ref{Csize_t}(0)
+    bytes = bytes_view(pqv)
+
+    unescaped_ptr = libpq_c.PQunescapeBytea(bytes, byte_length)
+
+    if unescaped_ptr == C_NULL
+        error("Could not unescape byte sequence $(String(bytes))")
+    end
+
+    unescaped_vec = copy(unsafe_wrap(Vector{UInt8}, unescaped_ptr, byte_length[]))
+
+    libpq_c.PQfreemem(unescaped_ptr)
+
+    return unescaped_vec
+end
+
+## bool
+# TODO: check whether we ever need this or if PostgreSQL always gives t or f
+_DEFAULT_TYPE_MAP[:bool] = Bool
+const BOOL_TRUE = r"^\s*(t|true|y|yes|on|1)\s*$"i
+const BOOL_FALSE = r"^\s*(f|false|n|no|off|0)\s*$"i
+function Base.parse(::Type{Bool}, pqv::PQValue{PQ_SYSTEM_TYPES[:bool]})
+    str = string_view(pqv)
+
+    if ismatch(BOOL_TRUE, str)
+        return true
+    elseif ismatch(BOOL_FALSE, str)
+        return false
+    else
+        error("\"$str\" is not a valid boolean")
+    end
+end
+
+## dates and times
+# ISO, YMD
+_DEFAULT_TYPE_MAP[:timestamp] = DateTime
+const TIMESTAMP_FORMAT = dateformat"y-m-d HH:MM:SS.s"  # .s is optional here
+function Base.parse(::Type{DateTime}, pqv::PQValue{PQ_SYSTEM_TYPES[:timestamp]})
+    str = string_view(pqv)
+
+    if str == "infinity"
+        return typemax(DateTime)
+    elseif str == "-infinity"
+        return typemin(DateTime)
+    end
+
+    return parse(DateTime, str, TIMESTAMP_FORMAT)
+end
+
+# ISO, YMD
+_DEFAULT_TYPE_MAP[:timestamptz] = ZonedDateTime
+const TIMESTAMPTZ_FORMATS = (
+    dateformat"y-m-d HH:MM:SSz",
+    dateformat"y-m-d HH:MM:SS.sz",
+    dateformat"y-m-d HH:MM:SS.ssz",
+    dateformat"y-m-d HH:MM:SS.sssz",
+    dateformat"y-m-d HH:MM:SS.ssssz",
+    dateformat"y-m-d HH:MM:SS.sssssz",
+    dateformat"y-m-d HH:MM:SS.ssssssz",
+)
```

Comment:
I'm sorry for this 😞

Reply:
Just let me know when the tryparse fix is tagged, and then if this becomes usable as one format :)

Reply:
`tryparse` fix is tagged on TimeZones.jl v0.6.0. Awaiting METADATA: https://github.com/JuliaLang/METADATA.jl/pull/12930